cmd/callVariants/testVariant.go

```diff
@@ -0,0 +1,359 @@
+package main
+
+import (
+	"fmt"
+	"github.com/vertgenlab/gonomics/chromInfo"
+	"github.com/vertgenlab/gonomics/dna"
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/fasta"
+	"github.com/vertgenlab/gonomics/numbers"
+	"github.com/vertgenlab/gonomics/sam"
+	"github.com/vertgenlab/gonomics/vcf"
+	"sort"
+	"strings"
+)
+
+// getVariant finds variants between the experimental and normal samples with p < maxP via fishers exact test.
+// returns a vcf record for the input position and a bool that is false if the returned variant should be discarded.
+func getVariant(exp, norm []sam.Pile, ref *fasta.Seeker, chroms []chromInfo.ChromInfo, maxP float64, minAf float64, minCoverage int) (vcf.Vcf, bool) {
+	var hasExp, hasNorm bool
+	var warnings []string
+	for i := range exp {
+		if exp[i].RefIdx != -1 { // RefIdx == -1 is mark for no data
+			hasExp = true
+		}
+	}
+	if !hasExp { // return empty struct if there are no experimental samples for the input site
+		return vcf.Vcf{}, false
+	}
+	for i := range norm {
+		if norm[i].RefIdx != -1 {
+			hasNorm = true
+		}
+	}
+	if !hasNorm {
+		warnings = append(warnings, "NO_NORMAL")
+	}
+
+	// get a background allele count for fishers exact testing
+	var bkgd sam.Pile
+	if hasNorm { // if normals are present, then use the summed normals for comparison
+		bkgd = sumPiles(norm)
+	} else { // if no normals are present then use experimental samples for comparison
+		bkgd = sumPiles(exp)
+	}
+
+	// pull the base before, as well as the current base in case an anchor is need for an indel
+	var seekStart, seekEnd int // Note that the Pos in pile is 1-base
+	seekStart = int(bkgd.Pos) - 2
+	seekEnd = int(bkgd.Pos)
+	if seekStart == -1 {
+		seekStart = 0 // TODO get some logic to add the base after a start-of-chromosome deletion for anchor
```

Comment:
This can also be included in our code review discussion about vcf handling the deletion of the first ref base.